### DISCUSSION TITLE
Add "Checkout and pull" option to menu for a branch

### DISCUSSION THREAD

**Discussion by richsidney** [gitextensions/gitextensions#9631] [Q&A] [answered]
Often I have to check out a branch and immediately pull from origin to start my work. It would be helpful to be able to do this with a singe click

**Comment by RussKie** [comment, marked as answer]
The left menu has a number of options:
![image](https://user-images.githubusercontent.com/4403806/136637823-8a9e1af0-92c0-4b2d-aad1-89a9bcc73947.png)

If those don't suit your needs you can write a script (Settings > Scripts).
